discord/ext/bridge/core.py

```diff
@@ -22,98 +22,74 @@
 FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 DEALINGS IN THE SOFTWARE.
 """
+
+import inspect
 from typing import Any, List, Union
-import asyncio
 
 import discord.commands.options
-from discord.commands import Option, SlashCommand
-from discord.enums import SlashCommandOptionType
-
-from ..commands import AutoShardedBot as ExtAutoShardedBot
-from ..commands import BadArgument
-from ..commands import Bot as ExtBot
+from discord import SlashCommandOptionType, Attachment, Option, SlashCommand, SlashCommandGroup
+from ...utils import get
+from ..commands.converter import _convert_to_bool, run_converters
 from ..commands import (
+    Context,
     Command,
+    Group,
     Converter,
     GuildChannelConverter,
     RoleConverter,
     UserConverter,
+    BadArgument,
+    Bot as ExtBot,
 )
 
-__all__ = ("BridgeCommand", "bridge_command", "BridgeExtCommand", "BridgeSlashCommand")
-
-from ...utils import get
-from ..commands.converter import _convert_to_bool
-
-
-class BridgeSlashCommand(SlashCommand):
-    """
-    A subclass of :class:`.SlashCommand` that is used to implement bridge commands.
-    """
-
-    ...
-
-
-class BridgeExtCommand(Command):
-    """
-    A subclass of :class:`.ext.commands.Command` that is used to implement bridge commands.
-    """
-
-    ...
 
+__all__ = (
+    "BridgeCommand",
+    "BridgeCommandGroup",
+    "bridge_command",
+    "bridge_group",
+    "BridgeExtCommand",
+    "BridgeSlashCommand",
+    "map_to",
+)
 
-class BridgeCommand:
-    """
-    This is the base class for commands that are compatible with both traditional (prefix-based) commands and slash
-    commands.
 
-    Parameters
-    ----------
-    callback: Callable[[:class:`.BridgeContext`, ...], Awaitable[Any]]
-        The callback to invoke when the command is executed. The first argument will be a :class:`BridgeContext`,
-        and any additional arguments will be passed to the callback. This callback must be a coroutine.
-    kwargs: Optional[Dict[:class:`str`, Any]]
-        Keyword arguments that are directly passed to the respective command constructors.
-    """
+def filter_params(params, **kwargs):
+    for param, nparam in kwargs.items():
+        if param in params:
+            if nparam is None:
+                params.pop(param)
+            else:
+                kwargs[nparam] = params.pop(param)
```

Comment:
Could be moved to utils?

Reply:
Yeah

Reply:
If you could add a docstring explaining what this is and why it's useful :)